File: typescript/src/DiscountCalculator.ts
export class DiscountCalculator {
    public static calculateDiscount(orderTotal: number, customerType: string): number {
        if (customerType === 'VIP') {
            return orderTotal * 0.1;
        }

        return orderTotal < 100 ? 0 : orderTotal * 0.05;
    }
}

File: java/src/main/java/DiscountCalculator.java

public class DiscountCalculator {

    public static double calculateDiscount(double orderTotal, String customerType) {
        if("VIP".equals(customerType)){
            return orderTotal * 0.1;
        }

        return orderTotal < 100 ? 0 : orderTotal * 0.05;
    }

}
